fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -165,13 +165,17 @@ def self.keychain_path(name)
       # has changed for some users: https://github.com/fastlane/fastlane/issues/5649
       #
 
+      # Remove the ".keychain" at the end of the name
+      name.sub!(/\.keychain$/, "")
+
       keychain_paths = [
         File.join(Dir.home, 'Library', 'Keychains', name),
-        File.join(Dir.home, 'Library', 'Keychains', "#{name}-db"),
-        name,
-        "#{name}-db"
+        name
       ].map { |path| File.expand_path(path) }
 
+      # Transforms ["thing"] to ["thing", "thing-db", "thing.keychain", "thing-db.keychain"]
+      keychain_paths = keychain_paths.product(["", "-db"], ["", ".keychain"]).map(&:join)
```

Comment:
I think that having a list like you put in the example above is the best solution because it makes it explicit where we are looking for the keychain, and I think that changing that one line with the functional operators will cause more fallout than changing a line in the list above, also if it becomes more complicated, the `product` and `map($:join)` calls will be much harder to reason about, but I think if you would really like to keep that code in there, then yeah, a more descriptive comment would be helpful. 

I just worry about a case where using `product` doesn't make sense, like if there are permutations of a path that do not conveniently match up as they do now. For example, it could be the case that `#{name}-keychain.something` but not `#{name}.something` is possible and then using `product` will produce erroneous paths.

All this being said, I think its fine to leave it and deal with it down the road if anything comes up.